### PR TITLE
Include unnecessary columns pruning step during federated plan creation

### PR DESCRIPTION
**Pull request by sgrebnov**
Upgrade `datafusion-federation` crate to include the following improvement:
[Run optimize_projections as part of federated plan optimization](https://github.com/spiceai/datafusion-federation/pull/24)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -100,5 +100,7 @@ sqlite-federation = ["sqlite"]
 postgres-federation = ["postgres"]
 
 [patch.crates-io]
-datafusion-federation = { git = "https://github.com/spiceai/datafusion-federation.git", rev = "0aabd423ef8aea0ae05479331d1486e9ef8f8061" }
-duckdb = { git = "https://github.com/spiceai/duckdb-rs.git", rev = "5b98603705a381ceeb5cc371e4f606b7332b57ce" }
\ No newline at end of file
+datafusion-federation = { git = "https://github.com/spiceai/datafusion-federation.git", rev = "914bd0836baa6990c5d03f977e5e87fe5eeaf4d6" }
+duckdb = { git = "https://github.com/spiceai/duckdb-rs.git", rev = "5b98603705a381ceeb5cc371e4f606b7332b57ce" }
+# Temporary patch to address SessionConfig::new().with_optimize_projections_preserve_existing_projections(true), method not found in `SessionConfig` when using datafusion-federation
+datafusion = { git = "https://github.com/spiceai/datafusion.git", rev = "06969ee5af853f0c071a98683dc2b9fde71b81a9" }
```
Comment (sgrebnov) [inline Cargo.toml:106]: Spent some time trying to figure out why `datafusion` version patch defined in `datafusion-federation` is overridden when `datafusion-federation` is added as dependency (I don't see any patches defined in `datafusion-table-providers`), added temporary patch to unblock adding this functionality in release. Will review after that

File: Cargo.toml [hunk 2]
```diff
@@ -100,5 +100,14 @@ sqlite-federation = ["sqlite"]
 postgres-federation = ["postgres"]
 
 [patch.crates-io]
-datafusion-federation = { git = "https://github.com/spiceai/datafusion-federation.git", rev = "0aabd423ef8aea0ae05479331d1486e9ef8f8061" }
-duckdb = { git = "https://github.com/spiceai/duckdb-rs.git", rev = "5b98603705a381ceeb5cc371e4f606b7332b57ce" }
\ No newline at end of file
+datafusion-federation = { git = "https://github.com/spiceai/datafusion-federation.git", rev = "914bd0836baa6990c5d03f977e5e87fe5eeaf4d6" }
+duckdb = { git = "https://github.com/spiceai/duckdb-rs.git", rev = "5b98603705a381ceeb5cc371e4f606b7332b57ce" }
+
+# Temporary patch to address SessionConfig::new().with_optimize_projections_preserve_existing_projections(true), method not found in `SessionConfig` when using datafusion-federation
```
Comment (phillipleblanc) [inline Cargo.toml:106]:
```suggestion
```

File: Cargo.toml [hunk 3]
```diff
@@ -100,5 +100,13 @@ sqlite-federation = ["sqlite"]
...
 
 [patch.crates-io]
-datafusion-federation = { git = "https://github.com/spiceai/datafusion-federation.git", rev = "0aabd423ef8aea0ae05479331d1486e9ef8f8061" }
-duckdb = { git = "https://github.com/spiceai/duckdb-rs.git", rev = "5b98603705a381ceeb5cc371e4f606b7332b57ce" }
\ No newline at end of file
+datafusion-federation = { git = "https://github.com/spiceai/datafusion-federation.git", rev = "914bd0836baa6990c5d03f977e5e87fe5eeaf4d6" }
+duckdb = { git = "https://github.com/spiceai/duckdb-rs.git", rev = "5b98603705a381ceeb5cc371e4f606b7332b57ce" }
+
+datafusion = { git = "https://github.com/spiceai/datafusion.git", rev = "06969ee5af853f0c071a98683dc2b9fde71b81a9"}
+datafusion-common = { git = "https://github.com/spiceai/datafusion.git", rev = "06969ee5af853f0c071a98683dc2b9fde71b81a9" }
```
Comment (phillipleblanc) [inline Cargo.toml:107]:
```suggestion
```

File: Cargo.toml [hunk 4]
```diff
@@ -100,5 +100,13 @@ sqlite-federation = ["sqlite"]
...
+datafusion-execution = { git = "https://github.com/spiceai/datafusion.git", rev = "06969ee5af853f0c071a98683dc2b9fde71b81a9" }
```
Comment (phillipleblanc) [inline Cargo.toml:108]:
```suggestion
```

Comment (sgrebnov) [reply Cargo.toml]: Removed unnecessary dependencies

File: Cargo.toml (hunk 3 above)
Comment (sgrebnov) [reply Cargo.toml]: Removed unnecessary dependencies

**Pull request merged**